src/PointwiseFunctions/AnalyticData/GrMhd/SpecInitialData.cpp

```diff
@@ -280,10 +282,27 @@ template <size_t ThermodynamicDim>
 template <typename DataType>
 void SpecInitialData<ThermodynamicDim>::VariablesComputer<DataType>::operator()(
     const gsl::not_null<Scalar<DataType>*> electron_fraction,
-    const gsl::not_null<Cache*> /*cache*/,
+    const gsl::not_null<Cache*> cache,
     hydro::Tags::ElectronFraction<DataType> /*meta*/) const {
-  std::fill(electron_fraction->begin(), electron_fraction->end(),
-            constant_electron_fraction);
+  const auto& temperature =
+      cache->get_var(*this, hydro::Tags::Temperature<DataType>{});
+  const auto& rest_mass_density =
+      get<hydro::Tags::RestMassDensity<DataType>>(interpolated_data);
+  const size_t num_points = get_size(get(rest_mass_density));
+  for (size_t i = 0; i < num_points; ++i) {
+     if (electron_fraction_value.has_value()) {
```

Comment:
Why do this check inside the for loop? Here you are setting all grid points to the constant value `num_points` times. If you put the `if` outside the for loop and have the for look just in the `else` then you don't do the redundant work.